### PR TITLE
Sample: Tweet on upcoming and recently released movies

### PR DESCRIPTION
**Pull request by oshadajay**
## Purpose
Sample: Tweet on upcoming and recently released movies

Fixes: https://github.com/wso2-enterprise/choreo/issues/19711

### CODE REVIEW

File: tweet_on_upcoming_and_recently_released_movies/main.bal [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    // Generate a tweet about five movies using Azure OpenAI   
+    final text:Client azureOpenAI = check new (
+        config = {httpVersion: http:HTTP_1_1, auth: {apiKey: openAIToken}},
+        serviceUrl = serviceUrl
+    );
+    string prompt = "Instruction: Generate a creative and descriptive tweet about upcoming and recently related movies and please include the following. Movies : ";
+    int i = 0;
+    foreach var movie in result.results{
+        i = i + 1;
+        prompt += i.toString() + ". " + movie.title + " - " + movie.overview + " ";
```
Comment (JayaniH) [inline tweet_on_upcoming_and_recently_released_movies/main.bal:42]:
```suggestion
        prompt += string `${i.toString()}. ${movie.title} - ${movie.overview} `;
```

File: tweet_on_upcoming_and_recently_released_movies/main.bal [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        if i == 5 {
+            break;
+        }
+    }
+
+    text:Deploymentid_completions_body completionsBody = {
+        prompt,
+        max_tokens: 1000
```
Comment (JayaniH) [inline tweet_on_upcoming_and_recently_released_movies/main.bal:50]: I think twitter has a max character limit per tweet. We may need to consider it.

File: tweet_on_upcoming_and_recently_released_movies/main.bal [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    };
+    text:Inline_response_200 completion = check azureOpenAI->/deployments/[deploymentId]/completions.post(API_VERSION, completionsBody);
+    string? tweetContent = completion.choices[0].text;
+
+    // Tweet it out!
+    if tweetContent is string{
```
Comment (JayaniH) [inline tweet_on_upcoming_and_recently_released_movies/main.bal:56]:
```suggestion
    if tweetContent is string { 
```

File: tweet_on_upcoming_and_recently_released_movies/main.bal [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    text:Deploymentid_completions_body completionsBody = {
+        prompt,
+        max_tokens: 1000
+    };
+    text:Inline_response_200 completion = check azureOpenAI->/deployments/[deploymentId]/completions.post(API_VERSION, completionsBody);
+    string? tweetContent = completion.choices[0].text;
+
+    // Tweet it out!
+    if tweetContent is string{ 
+        twitter:Client twitter = check new (twitterConfig);
```
Comment (JayaniH) [inline tweet_on_upcoming_and_recently_released_movies/main.bal:57]: we have used the `final` keyword when initializing the other two clients. why not here?

File: tweet_on_upcoming_and_recently_released_movies/main.bal [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    themoviedb:InlineResponse2001 result = check moviedb->getUpcomingMovies();
+
+    // Generate a tweet about five movies using Azure OpenAI   
+    final text:Client azureOpenAI = check new (
+        config = {httpVersion: http:HTTP_1_1, auth: {apiKey: openAIToken}},
+        serviceUrl = serviceUrl
+    );
+    string prompt = "Instruction: Generate a creative and descriptive tweet about upcoming and recently related movies and please include the following. Movies : ";
+    int i = 0;
+    foreach var movie in result.results{
```
Comment (JayaniH) [inline tweet_on_upcoming_and_recently_released_movies/main.bal:40]:
```suggestion
    foreach var movie in result.results {
```

File: tweet_on_upcoming_and_recently_released_movies/main.bal [hunk 6]
```diff
@@ -0,0 +1,61 @@
+import ballerina/io;
+import ballerinax/themoviedb;
+import ballerinax/azure_openai.text;
+import ballerina/http;
+import ballerinax/twitter;
```
Comment (JayaniH) [inline tweet_on_upcoming_and_recently_released_movies/main.bal:5]: i think these should be arranged in alphabetical order and follow the guidelines in
https://learn-ballerina.github.io/best_practices/format_the_code.html#order-of-the-imports-in-a-bal-file

**Review by JayaniH — COMMENTED** [review]
Shall we rename the sample to indicate that we are using azure openai for this?
Also, let's make sure that the code is properly formatted (spaces etc.)

File: tweet_on_upcoming_and_recently_released_movies/main.bal (hunk 6 above)
Comment (oshadajay) [reply tweet_on_upcoming_and_recently_released_movies/main.bal]: Addressed in https://github.com/ballerina-guides/ai-samples/pull/63/commits/7c756cfcba039786cbdf07ee3fe1a498746b3d32.

File: tweet_on_upcoming_and_recently_released_movies/main.bal (hunk 5 above)
Comment (oshadajay) [reply tweet_on_upcoming_and_recently_released_movies/main.bal]: Addressed in https://github.com/ballerina-guides/ai-samples/pull/63/commits/7c756cfcba039786cbdf07ee3fe1a498746b3d32.

File: tweet_on_upcoming_and_recently_released_movies/main.bal (hunk 4 above)
Comment (oshadajay) [reply tweet_on_upcoming_and_recently_released_movies/main.bal]: Addressed in https://github.com/ballerina-guides/ai-samples/pull/63/commits/7c756cfcba039786cbdf07ee3fe1a498746b3d32.

File: tweet_on_upcoming_and_recently_released_movies/main.bal (hunk 3 above)
Comment (oshadajay) [reply tweet_on_upcoming_and_recently_released_movies/main.bal]: Addressed in https://github.com/ballerina-guides/ai-samples/pull/63/commits/7c756cfcba039786cbdf07ee3fe1a498746b3d32.

File: tweet_on_upcoming_and_recently_released_movies/main.bal (hunk 2 above)
Comment (oshadajay) [reply tweet_on_upcoming_and_recently_released_movies/main.bal]: Addressed in https://github.com/ballerina-guides/ai-samples/pull/63/commits/7c756cfcba039786cbdf07ee3fe1a498746b3d32.

File: tweet_on_upcoming_and_recently_released_movies/main.bal (hunk 1 above)
Comment (oshadajay) [reply tweet_on_upcoming_and_recently_released_movies/main.bal]: Addressed in https://github.com/ballerina-guides/ai-samples/pull/63/commits/7c756cfcba039786cbdf07ee3fe1a498746b3d32.

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    final text:Client azureOpenAI = check new (
+        config = {httpVersion: http:HTTP_1_1, auth: {apiKey: openAIToken}},
+        serviceUrl = serviceUrl
+    );
+    string prompt = "Instruction: Generate a creative and short tweet about upcoming and recently related movies and please include the following. Movies : ";
+    int i = 0;
+    foreach var movie in result.results {
+        i = i + 1;
+        prompt += string `${i.toString()}. ${movie.title} - ${movie.overview} `;
+        if i == 5 {
```
Comment (malithjayasinghe) [inline tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal:44]: Define this as a const

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+            break;
+        }
+    }
+
+    text:Deploymentid_completions_body completionsBody = {
+        prompt,
+        max_tokens: 1000
```
Comment (malithjayasinghe) [inline tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal:51]: define this as a const

**Review by malithjayasinghe — CHANGES_REQUESTED** [review]

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+};
+
+public function main() returns error? {
+    // Get information on upcoming and recently released movies from TMDB
+    final themoviedb:Client moviedb = check new themoviedb:Client({apiKey : moviedbApiKey});
+    themoviedb:InlineResponse2001 result = check moviedb->getUpcomingMovies();
+
+    // Generate a tweet about five movies using Azure OpenAI   
+    final text:Client azureOpenAI = check new (
+        config = {httpVersion: http:HTTP_1_1, auth: {apiKey: openAIToken}},
```
[the remark this answers is not in the document]
Comment (oshadajay) [reply tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal]: Addressed in https://github.com/ballerina-guides/ai-samples/pull/63/commits/3ad5e3955599d64dd791ba6c368ca15db99a7121.

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal (hunk 8 above)
Comment (oshadajay) [reply tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal]: Addressed in https://github.com/ballerina-guides/ai-samples/pull/63/commits/3ad5e3955599d64dd791ba6c368ca15db99a7121.

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal (hunk 7 above)
Comment (oshadajay) [reply tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal]: Addressed in https://github.com/ballerina-guides/ai-samples/pull/63/commits/3ad5e3955599d64dd791ba6c368ca15db99a7121.

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/README.md [hunk 10]
```diff
@@ -0,0 +1,40 @@
+# Tweet on upcoming and recently released movies using Azure OpenAI
+
+## Use Case
+Obtaines information on few upcoming and recently released movies from [TMDB](https://www.themoviedb.org/), generates a creative tweet using Azure OpenAI GPT3 models, and tweets it out.
```
Comment (anupama-pathirage) [inline tweet_on_upcoming_and_recently_released_movies_using_azure_openai/README.md:4]:
```suggestion
Obtain information on a few upcoming and recently released movies from [TMDB](https://www.themoviedb.org/), generate a creative tweet using Azure OpenAI GPT3 models, and tweet it out.  
```

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+    foreach int i in 1 ... NO_OF_MOVIES {
+        var movie = moviedbRes.results[i - 1];
+        prompt += string `${i.toString()}. ${movie.title} `;  
+    }
+
+    text:Deploymentid_completions_body completionsBody = {
+        prompt,
+        max_tokens: MAX_TOKENS
+    };
+    text:Inline_response_200 completion = check azureOpenAI->/deployments/[deploymentId]/completions.post(API_VERSION, completionsBody);
```
Comment (anupama-pathirage) [inline tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal:51]: Break the line at max of 120 chars for readability.

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+    final themoviedb:Client moviedb = check new themoviedb:Client({apiKey : moviedbApiKey});
+    themoviedb:InlineResponse2001 moviedbRes = check moviedb->getUpcomingMovies();
+
+    // Generate a creative tweet using Azure OpenAI   
+    final text:Client azureOpenAI = check new (
+        config = {auth: {apiKey: openAIToken}},
+        serviceUrl = serviceUrl
+    );
+
+    string prompt = "Instruction: Generate a creative and short tweet below 250 characters about the following upcoming and recently released movies. Movies: ";
```
Comment (anupama-pathirage) [inline tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal:41]: Break the line at max of 120 chars

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+    accessTokenSecret: twitterAccessTokenSecret
+};
+
+public function main() returns error? {
+    // Get information on upcoming and recently released movies from TMDB
+    final themoviedb:Client moviedb = check new themoviedb:Client({apiKey : moviedbApiKey});
+    themoviedb:InlineResponse2001 moviedbRes = check moviedb->getUpcomingMovies();
+
+    // Generate a creative tweet using Azure OpenAI   
+    final text:Client azureOpenAI = check new (
```
Comment (anupama-pathirage) [inline tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal:36]: Can move this initialization just before the invocation at line 51

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+        config = {auth: {apiKey: openAIToken}},
+        serviceUrl = serviceUrl
+    );
+
+    string prompt = "Instruction: Generate a creative and short tweet below 250 characters about the following upcoming and recently released movies. Movies: ";
+    foreach int i in 1 ... NO_OF_MOVIES {
+        var movie = moviedbRes.results[i - 1];
```
Comment (anupama-pathirage) [inline tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal:43]:
```suggestion

```

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+        return error("Failed to generate a tweet on upcoming and recently released movies.");
+    }
+
+    if tweetContent.length() > MAX_TWEET_LENGTH {
+        return error("The generated tweet exceeded the maximum supported character length.");
+    } 
+
+    // Tweet it out!
+    final twitter:Client twitter = check new (twitterConfig);
+    var tweet = check twitter->tweet(tweetContent);
```
Comment (anupama-pathirage) [inline tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal:64]:
```suggestion
    twitter:Tweet tweet = check twitter->tweet(tweetContent);
```

Comment (oshadajay) [reply tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal]: Addressed in https://github.com/ballerina-guides/ai-samples/pull/63/commits/f7ce252feadcc1c2f7f19f7b18a3f078cbe22927.

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+    // Generate a creative tweet using Azure OpenAI   
+    final text:Client azureOpenAI = check new (
+        config = {auth: {apiKey: openAIToken}},
+        serviceUrl = serviceUrl
+    );
+
+    string prompt = "Instruction: Generate a creative and short tweet below 250 characters about the following upcoming and recently released movies. Movies: ";
+    foreach int i in 1 ... NO_OF_MOVIES {
+        var movie = moviedbRes.results[i - 1];
+        prompt += string `${i.toString()}. ${movie.title} `;
```
[the remark this answers is not in the document]
Comment (oshadajay) [reply tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal]: Addressed in https://github.com/ballerina-guides/ai-samples/pull/63/commits/f7ce252feadcc1c2f7f19f7b18a3f078cbe22927.

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal (hunk 14 above)
Comment (oshadajay) [reply tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal]: Addressed in https://github.com/ballerina-guides/ai-samples/pull/63/commits/f7ce252feadcc1c2f7f19f7b18a3f078cbe22927.

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal (hunk 13 above)
Comment (oshadajay) [reply tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal]: Addressed in https://github.com/ballerina-guides/ai-samples/pull/63/commits/f7ce252feadcc1c2f7f19f7b18a3f078cbe22927.

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal (hunk 12 above)
Comment (oshadajay) [reply tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal]: Addressed in https://github.com/ballerina-guides/ai-samples/pull/63/commits/f7ce252feadcc1c2f7f19f7b18a3f078cbe22927.

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal (hunk 11 above)
Comment (oshadajay) [reply tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal]: Addressed in https://github.com/ballerina-guides/ai-samples/pull/63/commits/f7ce252feadcc1c2f7f19f7b18a3f078cbe22927.

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/README.md [hunk 17]
```diff
@@ -0,0 +1,40 @@
...
+* Twitter developer account
+
+## Setting up Azure OpenAI service
+1. Follow the instructions given in [Azure Documentation](https://learn.microsoft.com/en-us/azure/cognitive-services/openai/how-to/create-resource?pivots=web-portal) to create an Azure OpenAI resource and deploy a text completion model.
+2. In the Azure Portal, go to **Keys and Endpoint** in the **Resource Management** section of the resource and obtain the service URL (**Endpoint**) and API key (either **KEY1** or **KEY2**).
+3. Go to **Deployments** in the **Resource Management** section and obtain the name of the text completion model which was deployed in step 1.
+
+## Setting up TMDB account
+1. Sign up and log into [TMDB](https://www.themoviedb.org/login).
+2. Click `Settings` and then click `API` tab in the left side bar.
```
[the remark this answers is not in the document]
Comment (oshadajay) [reply tweet_on_upcoming_and_recently_released_movies_using_azure_openai/README.md]: Addressed in https://github.com/ballerina-guides/ai-samples/pull/63/commits/f7ce252feadcc1c2f7f19f7b18a3f078cbe22927.

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/README.md [hunk 18]
```diff
@@ -0,0 +1,40 @@
...
+* Azure OpenAI service
+* TMDB account
+* Twitter developer account
+
+## Setting up Azure OpenAI service
+1. Follow the instructions given in [Azure Documentation](https://learn.microsoft.com/en-us/azure/cognitive-services/openai/how-to/create-resource?pivots=web-portal) to create an Azure OpenAI resource and deploy a text completion model.
+2. In the Azure Portal, go to **Keys and Endpoint** in the **Resource Management** section of the resource and obtain the service URL (**Endpoint**) and API key (either **KEY1** or **KEY2**).
+3. Go to **Deployments** in the **Resource Management** section and obtain the name of the text completion model which was deployed in step 1.
+
+## Setting up TMDB account
```
[the remark this answers is not in the document]
Comment (oshadajay) [reply tweet_on_upcoming_and_recently_released_movies_using_azure_openai/README.md]: Addressed in https://github.com/ballerina-guides/ai-samples/pull/63/commits/f7ce252feadcc1c2f7f19f7b18a3f078cbe22927.

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/README.md (hunk 10 above)
Comment (oshadajay) [reply tweet_on_upcoming_and_recently_released_movies_using_azure_openai/README.md]: Addressed in https://github.com/ballerina-guides/ai-samples/pull/63/commits/f7ce252feadcc1c2f7f19f7b18a3f078cbe22927.

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal [hunk 19]
```diff
@@ -0,0 +1,67 @@
...
+public function main() returns error? {
+    // Get information on upcoming and recently released movies from TMDB
+    final themoviedb:Client moviedb = check new themoviedb:Client({apiKey: moviedbApiKey});
+    themoviedb:InlineResponse2001 moviedbRes = check moviedb->getUpcomingMovies();
+
+    // Generate a creative tweet using Azure OpenAI   
+    string prompt = "Instruction: Generate a creative and short tweet below 250 characters about the following " +
+    "upcoming and recently released movies. Movies: ";
+    foreach int i in 1 ... NO_OF_MOVIES {
+        prompt += string `${i.toString()}. ${moviedbRes.results[i - 1].title} `;
```
Comment (anupama-pathirage) [inline tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal:39]: Do we need `toString` for `i` here?

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal [hunk 20]
```diff
@@ -0,0 +1,67 @@
...
+    final text:Client azureOpenAI = check new (
+        config = {auth: {apiKey: openAIToken}},
+        serviceUrl = serviceUrl
+    );
+    text:Inline_response_200 completion = check azureOpenAI->/deployments/[deploymentId]/completions.post(
+        API_VERSION, completionsBody
+    );
+    string? tweetContent = completion.choices[0].text;
+
+    if tweetContent !is string {
```
Comment (anupama-pathirage) [inline tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal:55]: check for nil. i.e. `if tweetContent is ()`

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal [hunk 21]
```diff
@@ -0,0 +1,67 @@
...
+    apiKey: twitterApiKey,
+    apiSecret: twitterApiSecret,
+    accessToken: twitterAccessToken,
+    accessTokenSecret: twitterAccessTokenSecret
+};
+
+public function main() returns error? {
+    // Get information on upcoming and recently released movies from TMDB
+    final themoviedb:Client moviedb = check new themoviedb:Client({apiKey: moviedbApiKey});
+    themoviedb:InlineResponse2001 moviedbRes = check moviedb->getUpcomingMovies();
```
Comment (anupama-pathirage) [inline tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal:33]:
```suggestion
    themoviedb:InlineResponse2001 upcomingMovies = check moviedb->getUpcomingMovies();
```

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal (hunk 19 above)
Comment (anupama-pathirage) [inline tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal:39]:
```suggestion
        prompt += string `${i.toString()}. ${upcomingMovies.results[i - 1].title} `;
```

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal [hunk 22]
```diff
@@ -0,0 +1,67 @@
...
+const MAX_TOKENS = 1000;
+const NO_OF_MOVIES = 5;
+
+// Twitter API client configuration
+twitter:ConnectionConfig twitterConfig = {
+    apiKey: twitterApiKey,
+    apiSecret: twitterApiSecret,
+    accessToken: twitterAccessToken,
+    accessTokenSecret: twitterAccessTokenSecret
+};
```
Comment (anupama-pathirage) [inline tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal:28]: Can replace with below line and remove line 12:15 and line 23:27

```ballerina
configurable twitter:ConnectionConfig twitterConfig = ?;
```

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/README.md [hunk 23]
```diff
@@ -0,0 +1,40 @@
...
+## Configuration
+Create a file called `Config.toml` at the root of the project.
+
+### Config.toml
+```
+moviedbApiKey = "<TMDB_API_KEY>
+openAIToken = "<AZURE_OPENAI_API_KEY>"
+serviceUrl = "<AZURE_OPENAI_ENDPOINT>"
+deploymentId = "<AZURE_MODEL_DEPLOYMENT_NAME>"
+twitterApiKey = "<TWITTER_API_KEY>"
```
Comment (anupama-pathirage) [inline tweet_on_upcoming_and_recently_released_movies_using_azure_openai/README.md:33]:
```suggestion
[twitterConfig]
twitterApiKey = "<TWITTER_API_KEY>"
```
Need the above change due to comment https://github.com/ballerina-guides/ai-samples/pull/63/files#r1157568177

Comment (oshadajay) [reply tweet_on_upcoming_and_recently_released_movies_using_azure_openai/README.md]: Addressed in https://github.com/ballerina-guides/ai-samples/pull/63/commits/51d5361f5867dd852237e579e3c1e9938ebaf8ae.

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal (hunk 22 above)
Comment (oshadajay) [reply tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal]: Addressed in https://github.com/ballerina-guides/ai-samples/pull/63/commits/51d5361f5867dd852237e579e3c1e9938ebaf8ae.

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal (hunk 19 above)
Comment (oshadajay) [reply tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal]: Addressed in https://github.com/ballerina-guides/ai-samples/pull/63/commits/51d5361f5867dd852237e579e3c1e9938ebaf8ae.

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal (hunk 21 above)
Comment (oshadajay) [reply tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal]: Addressed in https://github.com/ballerina-guides/ai-samples/pull/63/commits/51d5361f5867dd852237e579e3c1e9938ebaf8ae.

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal (hunk 20 above)
Comment (oshadajay) [reply tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal]: Addressed in https://github.com/ballerina-guides/ai-samples/pull/63/commits/51d5361f5867dd852237e579e3c1e9938ebaf8ae.

File: tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal (hunk 19 above)
Comment (oshadajay) [reply tweet_on_upcoming_and_recently_released_movies_using_azure_openai/main.bal]: Addressed in https://github.com/ballerina-guides/ai-samples/pull/63/commits/51d5361f5867dd852237e579e3c1e9938ebaf8ae.

**Review by anupama-pathirage — APPROVED** [review]

**Review by malithjayasinghe — APPROVED** [review]

**Pull request merged**